File: src/main/java/com/wxl/dyttcrawler/domain/DyttMovie.java
package com.wxl.dyttcrawler.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

/**
 * Create by wuxingle on 2020/5/2
 * 电影详情
 */
@Data
public class DyttMovie implements Serializable {

    private static final long serialVersionUID = 4872108258435579856L;

    private String id;

    // 标题
    private String title;

    // 地址链接
    private String url;

    // 封面图片地址
    private String picUrl;

    // 译名
    private List<String> translateNames;

    // 片名
    private String name;

    // 年代
    private Integer year;

    // 产地
    private List<String> originPlace;

    // 类别
    private List<String> category;

    // 语言
    private List<String> language;

    // 字幕
    private String words;

    // 豆瓣评分
    private Double score;

    // 评分人数
    private Integer scoreNums;

    // 上映时间
    private List<DyttReleaseDate> releaseDates;

    // 导演
    private List<String> director;

    // 编剧
    private List<String> screenwriter;

    // 主演
    private List<String> act;

    // 标签
    private List<String> tags;

    // 获奖情况
    private List<String> awards;

    // 简介
    private String desc;

    // 发布时间
    @JsonFormat(pattern = "yyyy-MM-dd")
    private Date publishDate;

    // 下载地址
    private List<String> downLinks;

    // 文件大小 gb
    private Double fileSize;

    // 该文档更新时间
    @JsonFormat
    private Date updateTime;
}

File: src/main/kotlin/com/wxl/dyttcrawler/domain/DyttMovie.kt
package com.wxl.dyttcrawler.domain

import com.fasterxml.jackson.annotation.JsonFormat
import java.util.*

/**
 * Create by wuxingle on 2021/10/12
 * 电影详情
 */
data class DyttMovie(
    var id: String? = null,

    // 标题
    var title: String? = null,

    // 地址链接
    var url: String? = null,

    // 封面图片地址
    var picUrl: String? = null,

    // 译名
    var translateNames: List<String>? = null,

    // 片名
    var name: String? = null,

    // 年代
    var year: Int? = null,

    // 产地
    var originPlace: List<String>? = null,

    // 类别
    var category: List<String>? = null,

    // 语言
    var language: List<String>? = null,

    // 字幕
    var words: String? = null,

    // 豆瓣评分
    var score: Double? = null,

    // 评分人数
    var scoreNums: Int? = null,

    // 上映时间
    var releaseDates: List<DyttReleaseDate>? = null,

    // 导演
    var director: List<String>? = null,

    // 编剧
    var screenwriter: List<String>? = null,

    // 主演
    var act: List<String>? = null,

    // 标签
    var tags: List<String>? = null,

    // 获奖情况
    var awards: List<String>? = null,

    // 简介
    var desc: String? = null,

    // 发布时间
    @JsonFormat(pattern = "yyyy-MM-dd")
    var publishDate: Date? = null,

    // 下载地址
    var downLinks: List<String>? = null,

    // 文件大小 gb
    var fileSize: Double? = null,

    // 该文档更新时间
    @JsonFormat
    var updateTime: Date? = null

)